### DISCUSSION TITLE
Query logs

### DISCUSSION THREAD

**Discussion by xinyuluo** [osm-search/Nominatim#2554] [Q&A] [answered]
Hi! I installed Nominatim on my own machine, and enabled logging setting to file in `.env` to export queries log to a file. I believe the default log contains the following information: timestamp of the request, time processing, number of results, type of request (search, reverse, etc.), and input query.

I wonder if it's possible and where to edit the information contained in this file, so as to add other information if needed (such as result json)? Thanks.

**Comment by lonvia** [comment, marked as answer]
There is currently no way to configure the format of the log file. If you are determined, you can have a look at [log.php](https://github.com/osm-search/Nominatim/blob/master/lib-php/log.php) which is responsible for query logging and make changes to the output there. There are two functions, one is called at the beginning of request processing, the other at the end. If you want to log entire results, then this will require some wider changes because that information is currently not handed in into the log functions.

**Reply by xinyuluo** [reply]
@lonvia thanks for fast reply. sorry, I think I'm looking at the wrong place as I didn't find log.php file. I only found, under website directory:
```
$ ls
deletable.php  details.php  lookup.php  polygons.php  reverse.php  search.php  status.php  test.php
```
Can you please direct me to the right directory for log.php file? Thanks!

**Reply by lonvia** [reply]
The response contains a link to the file.
